bsp_diff/caas/device/intel/civ/host/vm-manager/0046-Multi-Android-VM-support.patch

```diff
@@ -0,0 +1,347 @@
++        $CIV_WORK_DIR/scripts/setup_civ_ini.sh -v $1 -p $2 -n $3
+ }
+ 
+ function create_aaf_dir() {
+         mkdir -p $CIV_WORK_DIR/scripts/aaf
+ }
+ 
++function create_vm_dir() {
++	echo "Dir: $CIV_WORK_DIR/$VM_NAME"
++	mkdir $CIV_WORK_DIR/$VM_NAME
++}
++
++function copy_files_for_vm() {
++    echo "Copying file: $CIV_WORK_DIR/$VM_NAME"
++    mkdir $CIV_WORK_DIR/$VM_NAME/scripts
++    cp -r $CIV_WORK_DIR/scripts/aaf $CIV_WORK_DIR/$VM_NAME/scripts/
++    cp OVMF.fd $CIV_WORK_DIR/$VM_NAME/
++    ls -la $CIV_WORK_DIR/$VM_NAME/
++    cp $CIV_WORK_DIR/scripts/rpmb_dev $CIV_WORK_DIR/$VM_NAME/scripts/
++}
+ 
+ #-------------    main processes    -------------
+ 
+@@ -618,22 +656,21 @@ check_network
+ check_kernel_version
+ check_sriov_setup
+ 
+-ubu_changes_require
+-ubu_install_qemu_gvt
+-ubu_build_ovmf_gvt
+-ubu_enable_host_gvt
+-ubu_update_fw
++if [ $skip_initial_host_setup != true ]; then
++    ubu_changes_require
++    ubu_install_qemu_gvt
++    ubu_enable_host_gvt
++    ubu_update_fw
++    ubu_build_ovmf_gvt
++fi
+ 
+-install_vm_manager
++create_vm_dir
+ create_aaf_dir
+-setup_civ_ini
+-
+-prepare_required_scripts
+-ubu_install_swtpm
+-ubu_install_libssl
+-ubu_update_bt_fw
+-set_sleep_inhibitor
++setup_civ_ini $VSOCK_ID $ADB_PORT $VM_NAME
++copy_files_for_vm
+ 
+-ask_reboot
++if [ $skip_initial_host_setup != true ]; then
++    ask_reboot
++fi
+ 
+ echo "Done: \"$(realpath $0) $@\""
+diff --git a/scripts/setup_host_multiple_android_vm.sh b/scripts/setup_host_multiple_android_vm.sh
+new file mode 100755
+index 0000000..894de87
+--- /dev/null
++++ b/scripts/setup_host_multiple_android_vm.sh
+@@ -0,0 +1,80 @@
++#!/bin/bash
++
++# Copyright (c) 2020 Intel Corporation.
++# All rights reserved.
++#
++# SPDX-License-Identifier: Apache-2.0
++
++set -eE
++
++#---------      Global variable     -------------------
++CIV_DIR=$(pwd)
++USER_HOME=$(getent passwd $SUDO_USER | cut -d: -f6)
++VM_NAME=Android_CIV
++VSOCK_ID=3
++NUMBER_OF_VM=1
++ADB_PORT=5555
++INITIAL_VM_NUM=1
++SETUP_HOST_DONE=false
++
++function setup_files() {
++    echo "Creating setup for VMs"
++    if [ -e $CIV_DIR/edk2 ]
++    then
++        SETUP_HOST_DONE=true
++    fi
++    new_dir=$CIV_DIR/$VM_NAME
++    if [ -n "$(ls -d ${new_dir}?)" ]; then
++	dir_arr=$(ls -d ${new_dir}?)
++	directories=( $dir_arr )
++	last_vm=${directories[-1]}
++	final_vm_name="${last_vm##*/}"
++	last_character=${final_vm_name: -1}
++	last_character=$((last_character+1))
++	ADB_PORT=$((5555+(last_character*2)))
++	VSOCK_ID=$((3+last_character))
++	INITIAL_VM_NUM=$last_character
++    fi
++
++    for (( i=$INITIAL_VM_NUM; i<$NUMBER_OF_VM+$INITIAL_VM_NUM; i++ ))
++    do
++	echo "Folder name: $CIV_DIR/${VM_NAME}${i}"
++
++	if [ $SETUP_HOST_DONE = false ]; then
++            ./scripts/setup_host.sh -v 3 -b $ADB_PORT -n ${VM_NAME}${i}
++	    setup_host_done=y
++	else
++	    ADB_PORT=$((ADB_PORT+(i*2)))
++	    ./scripts/setup_host.sh -s -v $((i+2)) -b $ADB_PORT -n ${VM_NAME}${i}
++	fi
```

Comment:
Why run setup_host.sh for each VM created?
Can we just recreate the folder structure and copy the files?